sdk/python/feast/permissions/auth/oidc_token_parser.py

```diff
@@ -0,0 +1,106 @@
+import logging
+import os
+from unittest.mock import Mock
+
+import jwt
+from dotenv import load_dotenv
+from fastapi import Request
+from fastapi.security import OAuth2AuthorizationCodeBearer
+from jwt import PyJWKClient
+from starlette.authentication import (
+    AuthenticationError,
+)
+
+from feast.permissions.auth.token_parser import TokenParser
+from feast.permissions.user import User
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class OidcTokenParser(TokenParser):
+    """
+    A `TokenParser` to use an OIDC server to retrieve the user details.
+    # TODO RBAC: use auth configuration instead
+    Server settings are retrieved fro these environment variables:
+    - `OIDC_SERVER_URL`
+    - `REALM`
+    - `CLIENT_ID`
+    """
+
+    def __init__(self):
+        # TODO RBAC: use auth configuration instead
+        load_dotenv("./.env")
+
+        self._OIDC_SERVER_URL = os.getenv("OIDC_SERVER_URL")
+        self._REALM = os.getenv("REALM")
+        self._CLIENT_ID = os.getenv("CLIENT_ID")
+
+    async def _validate_token(self, access_token: str):
+        """
+        Validate the token extracted from the headrer of the user request against the OAuth2 server.
+        """
+        # FastAPI's OAuth2AuthorizationCodeBearer requires a Request type but actually uses only the headers field
+        # https://github.com/tiangolo/fastapi/blob/eca465f4c96acc5f6a22e92fd2211675ca8a20c8/fastapi/security/oauth2.py#L380
+        request = Mock(spec=Request)
+        request.headers = {"Authorization": f"Bearer {access_token}"}
+
+        oauth_2_scheme = OAuth2AuthorizationCodeBearer(
+            tokenUrl=f"{self._OIDC_SERVER_URL}/realms/{self._REALM}/protocol/openid-connect/token",
```

Comment:
I think we may need to get these url from the discovery url. These URLs may work well with keycloak but suspect if it can work with other oidc idps.

Reply:
Good point, we can take it for a next review (adding subtask in Jira)